text/0028-bill-of-materials.md

```diff
@@ -0,0 +1,105 @@
+# Implement a Bill of Materials Across Paketo
+
+## Summary
+
+A bill of materials (BOM) is a set of metadata that provides an account of the
+dependencies and tools that are both present in an application image, as well
+those used in the construction of that image. The availability of this data is
+a key value proposition of Cloud Native Buildpacks.
+
+As an implementation of Cloud Native Buildpacks, the Paketo project should
+support adding bill of materials metadata across the different components that
+go into an app image. This RFC serves as an intent to implement the BOM and
+defines the minimal set of key/values to be included.
+
+## Motivation
+
+A BOM can include an array of different metadata about application dependencies
+and buildpack dependencies such as SHAs, URIs, versions, license information,
+vulnerability identifiers, etc. This information is valuable for users who want
+an accurate inventory of everything in their final application image, or that
+may have been used to build it. Having access to that information makes it
+easier to audit for vulnerabilities and license information.
+
+The Paketo Java Buildpacks as well as a handful of other Paketo Buildpacks
+already make the BOM available. Another motivation for doing this work is to
+achieve a consistent user experience across the Paketo project.
+
+## Detailed Explanation
+
+The following Paketo components should have BOM metadata attached to them:
+
+### Stacks
+Stacks (such as those found in the [Paketo Stacks repository](https://github.com/paketo-buildpacks/stacks))
+should have BOM metadata that includes in-depth information on all of the OS
+level packages installed as part of the stack.
+
+The minimal set of metadata for stacks should be:
+```
+{
+  "name" : "<package name>",
+  "version" : "<package version>",
+  "arch" : "<compatible architecture>",
```

Comment:
What is the utility of `arch`?

Reply:
The processor architecture that the package is compatible, such as `amd64`.
While it's perhaps not the most valuable BoM field, it's information that we already currently surface for the stacks in the `io.paketo.stack.packages` label, so we thought we might as well surface more information than less.